API.md

```diff
@@ -0,0 +1,123 @@
+# Authmosphere API documentation
+
+## Mock tooling
+
+This tooling provides an abstraction to easily mock oAuth 2.0 (RFC) related endpoints.
+
+It helps to easily write integration tests without writing extensive boilerplate code and without disabling oAuth for tests.
+
+This tooling is based on Nock, a HTTP mocking library. For more information about Nock see [Nock documentation](https://github.com/node-nock/nock).
+
+### mockAccessTokenEndpoint
+
+Creates a __very basic__ mock of token endpoint as defined in [RFC 6749](https://tools.ietf.org/html/rfc6749).
+
+The mocked endpoint will return a token(Link to type) with the scopes specified in the request.
+
+* ⚠️ The mock does not validate the request
+* ⚠️ The mock holds a state that contains the created tokens
+* ⚠️ `cleanMock` resets the state and removes all nocks
+
+#### Usage
+
+```typescript
+mockAccessTokenEndpoint({
+  url: 'http://some.oauth.endpoint/access_token',
+  times: 1
+});
+```
+
+#### Signature
+
+`mockAccessTokenEndpoint(options) => nock.Scope`
+
+#### Arguments
+
+* `options`:
+  * `url: string` - url of the Token validation endpoint
+  * `times?: number` - defines number of calls the endpoint is mocked, default is `Number.MAX_SAFE_INTEGER`
+
```

Comment:
I think a horizontal line would help to separate the methods (it is difficult to see where one doc ends and the next begins currently)

Reply:
Or is there maybe some other way to emphasize the separation? Can also think about that later

Reply:
I thought about using HR as seperator between different parts of the documentation